6/test/run

```diff
@@ -0,0 +1,246 @@
+#!/bin/bash
+#
+# The 'run' performs a simple test that verifies that STI image.
+# The main focus here is to excersise the STI scripts.
+#
+# IMAGE_NAME specifies a name of the candidate image used for testing.
+# The image has to be available before this script is executed.
+#
+IMAGE_NAME=${IMAGE_NAME-openshift/nodejs-6-centos7-candidate}
+
+test_dir="$(readlink -zf $(dirname "${BASH_SOURCE[0]}"))"
+image_dir=$(readlink -zf ${test_dir}/..)
+cid_file=$(mktemp -u --suffix=.cid)
+
+# Since we built the candidate image locally, we don't want S2I attempt to pull
+# it from Docker hub
+s2i_args="--force-pull=false "
+
+# TODO: This should be part of the image metadata
+test_port=8080
+
+if [ "$DEBUG" != "" ]; then
+  set -x
+fi
+
+image_exists() {
+  docker inspect $1 &>/dev/null
+}
+
+container_exists() {
+  image_exists $(cat $cid_file)
+}
+
+container_ip() {
+  docker inspect --format="{{ .NetworkSettings.IPAddress }}" $(cat $cid_file)
+}
+
+container_logs() {
+  docker logs $(cat $cid_file)
+}
+
+run_s2i_build() {
+  s2i build ${s2i_args} file://${test_dir}/test-app ${IMAGE_NAME} ${IMAGE_NAME}-testapp
+}
+
+prepare() {
+  if ! image_exists ${IMAGE_NAME}; then
+    echo "ERROR: The image ${IMAGE_NAME} must exist before this script is executed."
+    exit 1
+  fi
+  # TODO: STI build require the application is a valid 'GIT' repository, we
+  # should remove this restriction in the future when a file:// is used.
+  pushd ${test_dir}/test-app >/dev/null
+  git init
+  git config user.email "build@localhost" && git config user.name "builder"
+  git add -A && git commit --no-gpg-sign -m "Sample commit"
+  popd >/dev/null
+}
+
+run_test_application() {
+  docker run --user=100001 --rm --cidfile=${cid_file} -p ${test_port} $1 ${IMAGE_NAME}-testapp
```

Comment:
Please do not publish container's port to the host - application in container is accessed by container IP, so this is not necessary for tests. And running tests with this can cause conflicts with other containers in host system using same port...